src/Bindings/RabbitMQAsyncCollector.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ
+{
+    public class RabbitMQAsyncCollector : IAsyncCollector<string>
+    {
+        private readonly RabbitMQContext context;
+        private readonly ConcurrentQueue<string> messages = new ConcurrentQueue<string>();
+        private readonly ConnectionFactory factory;
+        private readonly IConnection connection;
+        private readonly IModel channel;
+        private readonly QueueDeclareOk queue;
+
+        public RabbitMQAsyncCollector(RabbitMQContext context)
+        {
+            this.context = context;
+            this.factory = new ConnectionFactory() { HostName = context.Hostname };
+            this.connection = this.factory.CreateConnection();
+            this.channel = this.connection.CreateModel();
+            this.queue = CreateQueue(this.channel, this.context);
+        }
+
+        public Task AddAsync(string message, CancellationToken cancellationToken = default)
+        {
+            message = this.context.Message;
+            this.messages.Enqueue(message);

```

Comment:
In order to support all types, you need to convert object to byte[]. See https://stackoverflow.com/questions/1446547/how-to-convert-an-object-to-a-byte-array-in-c-sharp for examples.
cc @brettsam if he has other ideas